### PR TITLE
CTW-541 Adding firstparty search param

### PR DESCRIPTION
**Pull request by eric-gunther**
Adding `firstparty` search parameter when a supported resource type is used. *DO NOT MERGE THIS UNTIL ODS HAS CONFIRMED THAT REINDEXING IS COMPLETE*.

Supported resources defined in ODS [like this](https://github.com/zeus-health/ods-fhir-resources/pull/41/files).

### CODE REVIEW

File: src/fhir/search-helpers.ts [hunk 1]
```diff
@@ -80,10 +90,22 @@ export async function searchBuilderRecords<T extends ResourceTypeString>(
     ...UPI_TAGS,
   ];
   const builderTag = `${SYSTEM_ZUS_OWNER}|builder/${requestContext.builderId}`;
-  const params = mergeParams(searchParams, {
+  type FirstPartParams = {
```
Comment (mrosata) [inline src/fhir/search-helpers.ts:93]:
```suggestion
  type FirstPartyParams = {
```

File: src/fhir/search-helpers.ts [hunk 2]
```diff
@@ -80,10 +90,22 @@ export async function searchBuilderRecords<T extends ResourceTypeString>(
...
+    _tag: string[];
+    firstparty?: boolean;
+    "_tag:not"?: string[];
+  };
+  const firstPartyParams: FirstPartParams = {
```
Comment (mrosata) [inline src/fhir/search-helpers.ts:98]:
```suggestion
  const firstPartyParams: FirstPartyParams = {
```

**Review by mrosata — DISMISSED** [review]
left a couple suggestions for a name typo, otherwise lgtm

**Review by pmadria3 — COMMENTED** [review]
LGTM -> Mike already approved but took a peek at the PR

**Review by luthfib — DISMISSED** [review]

**Review by mrosata — APPROVED** [review]

**Pull request merged**